stdlib/io/src/main/java/org/ballerinalang/stdlib/io/nativeimpl/CloseWritableCharacterChannel.java

```diff
@@ -52,25 +49,14 @@
 
     public static Object close(Strand strand, ObjectValue channel) {
         CharacterChannel charChannel = (CharacterChannel) channel.getNativeData(IOConstants.CHARACTER_CHANNEL_NAME);
-        EventContext eventContext = new EventContext(new NonBlockingCallback(strand));
-        CloseCharacterChannelEvent closeEvent = new CloseCharacterChannelEvent(charChannel, eventContext);
-        Register register = EventRegister.getFactory().register(closeEvent,
-                                                                CloseWritableCharacterChannel::closeChannel);
-        eventContext.setRegister(register);
-        register.submit();
+        try {
+            charChannel.close();
+        } catch (ClosedChannelException e) {
+            return IOUtils.createError("channel already closed.");
+        } catch (IOException e) {
+            return IOUtils.createError(e.getMessage());
```

Comment:
Since in several places we are using `IOUtils.createError(e.getMessage());` Wouldn't it make sense to have a util function that takes a throwable?